tests/pytests/test_config.py

```diff
@@ -370,14 +371,45 @@ def testAllConfigCoord(env):
 
 @skip(cluster=False)
 def testInitConfigCoord():
-    def test_arg_num(arg_name, arg_value):
-        env = Env(moduleArgs=f'{arg_name} {arg_value}', noDefaultModuleArgs=True)
-        env.expect(config_cmd(), 'get', arg_name).equal([[arg_name, str(arg_value)]])
+
+    _test_arg_num('SEARCH_THREADS', 3)
+    _test_arg_num('CONN_PER_SHARD', 3)
+
+    def _testOSSGlobalPasswordConfig():
+        env = Env(moduleArgs='OSS_GLOBAL_PASSWORD 123456', noDefaultModuleArgs=True)
+        if env.env == 'existing-env':
+            env.skip()
+        env.expect(config_cmd(), 'get', 'OSS_GLOBAL_PASSWORD').equal([['OSS_GLOBAL_PASSWORD', 'Password: *******']])
         env.stop()
 
-    test_arg_num('SEARCH_THREADS', 3)
-    test_arg_num('CONN_PER_SHARD', 3)
+    # We test `OSS_GLOBAL_PASSWORD` manually since the getter obfuscates the value
+    _testOSSGlobalPasswordConfig()
+
+    _test_arg_str('OSS_ACL_USERNAME', 'default')
 
 @skip(cluster=False)
 def testImmutableCoord(env):
     env.expect(config_cmd(), 'set', 'SEARCH_THREADS').error().contains(not_modifiable)
+    env.expect(config_cmd(), 'set', 'OSS_GLOBAL_PASSWORD').error().contains(not_modifiable)
+    env.expect(config_cmd(), 'set', 'OSS_ACL_USERNAME').error().contains(not_modifiable)
+
+@skip(cluster=False)
+def testSetACLUsername():
+    """Tests that the OSS_ACL_USERNAME configuration is set correctly on module
+    load"""
+
+    # Setting the `OSS_ACL_USERNAME` configuration without the `OSS_GLOBAL_PASSWORD`
+    # the configuration should not do anything since we don't try to authenticate.
+    _test_arg_str('OSS_ACL_USERNAME', 'test')
+
+    # Set both the username and password. This should fail since we have no such
+    # user.
+    env = Env(moduleArgs='OSS_ACL_USERNAME test_user OSS_GLOBAL_PASSWORD 123456', noDefaultModuleArgs=True)
+
+    timeout = 3 # 5 seconds, more than enough for the an env to be up
+    try:
+        with TimeLimit(timeout):
+            env.cmd('FT.SEARCH', 'idx', '*')
+            env.assertTrue(False)
+    except Exception as e:
+        env.assertEqual(str(e), 'Timeout: operation timeout exceeded')
```

Comment:
I think tests that rely on timeouts tend to eventually become flaky.

Reply:
I agree it's not ideal, yet a hanging client is the current behavior of this situation, so until we don't return a proper error for this situation I have no other way to test this.
I can comment this part out for now if we don't want it to be part of our CI (that I can understand and agree with)